sympy/stats/joint_rv.py

```diff
@@ -163,3 +165,45 @@ def marginal_distribution(rv, *indices):
     if hasattr(prob_space.distribution, 'marginal_distribution'):
         return prob_space.distribution.marginal_distribution(indices, rv.symbol)
     return prob_space.marginal_distribution(*indices)
+
+class MarginalDistribution(Basic):
+
+    def __new__(cls, symbol, pdf, rvs=None):
+        symbol = sympify(symbol)
+        return Basic.__new__(cls, symbol, pdf, rvs)
+
+    def check(self):
+        pass
+
+    def set(self):
+        rvs = (i for i in random_symbols(self.args[1]))
+        return ProductSet((i.pspace.set for i in rvs))
+
+    def pdf(self, x):
+        expr = self.args[1]
```

Comment:
It might be a good idea to extract all the used arguments into variables with proper names. Apart from `args[1]` which refers to the expression, I cannot make out what the other two refer to.
In future, it makes it difficult for other contributors to work and modify the code.